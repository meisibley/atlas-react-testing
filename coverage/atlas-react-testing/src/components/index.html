
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for atlas-react-testing/src/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> atlas-react-testing/src/components</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">28.3% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>75/265</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">57.89% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>11/19</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">60% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/15</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">28.3% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>75/265</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="CoverArt.tsx"><a href="CoverArt.tsx.html">CoverArt.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file low" data-value="CurrentlyPlaying.tsx"><a href="CurrentlyPlaying.tsx.html">CurrentlyPlaying.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	</tr>

<tr>
	<td class="file high" data-value="FooBar.tsx"><a href="FooBar.tsx.html">FooBar.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	</tr>

<tr>
	<td class="file low" data-value="Footer.tsx"><a href="Footer.tsx.html">Footer.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	</tr>

<tr>
	<td class="file low" data-value="MusicPlayer.tsx"><a href="MusicPlayer.tsx.html">MusicPlayer.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="65" class="abs low">0/65</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="65" class="abs low">0/65</td>
	</tr>

<tr>
	<td class="file low" data-value="PlayControls.tsx"><a href="PlayControls.tsx.html">PlayControls.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="65" class="abs low">0/65</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="65" class="abs low">0/65</td>
	</tr>

<tr>
	<td class="file high" data-value="PlayListItem.tsx"><a href="PlayListItem.tsx.html">PlayListItem.tsx</a></td>
	<td data-value="86.36" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 86%"></div><div class="cover-empty" style="width: 14%"></div></div>
	</td>
	<td data-value="86.36" class="pct high">86.36%</td>
	<td data-value="22" class="abs high">19/22</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="86.36" class="pct high">86.36%</td>
	<td data-value="22" class="abs high">19/22</td>
	</tr>

<tr>
	<td class="file low" data-value="Playlist.tsx"><a href="Playlist.tsx.html">Playlist.tsx</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="27" class="abs low">0/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="27" class="abs low">0/27</td>
	</tr>

<tr>
	<td class="file high" data-value="SongTitle.tsx"><a href="SongTitle.tsx.html">SongTitle.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="VolumeControl.tsx"><a href="VolumeControl.tsx.html">VolumeControl.tsx</a></td>
	<td data-value="89.28" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 89%"></div><div class="cover-empty" style="width: 11%"></div></div>
	</td>
	<td data-value="89.28" class="pct high">89.28%</td>
	<td data-value="28" class="abs high">25/28</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="89.28" class="pct high">89.28%</td>
	<td data-value="28" class="abs high">25/28</td>
	</tr>

<tr>
	<td class="file empty" data-value="types.ts"><a href="types.ts.html">types.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="1" class="abs empty">1/1</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="1" class="abs empty">1/1</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-09-30T20:04:28.295Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    