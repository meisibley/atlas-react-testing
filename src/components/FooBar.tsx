//FooBar.tsx
import { useState } from "react";

export function FooBar() {
    const [showFoo, setShowFoo] = useState(true);
    const [showBar, setShowBar] = useState(false);

    return (
        <div>
            {showFoo && <span>Foo</span>}
            {showBar && <span>Bar</span>}
            <button onClick={() => setShowFoo(!showFoo)}>
                Toggle Foo
            </button>
            <button onClick={() => setShowBar(!showBar)}>
                Toggle Bar
            </button>
        </div>
    );
}
