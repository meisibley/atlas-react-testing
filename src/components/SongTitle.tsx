import React from 'react';

type SongTitleProps = {
    title: string;
    artist: string;
};

export default function SongTitle({ title, artist }: SongTitleProps) {
    return (
        <div>
            <p className="text-black text-xl font-bold mt-6">{title}</p>
            <p className="text-gray font-semibold">{artist}</p>
        </div>
    );
};

// Another way to write it:
// interface SongTitleProps {
//     title: string;
//     artist: string;
// }
// const SongTitle: React.FC<SongTitleProps> = ( {title, artist}) => {
//     return (
//         <div>
//             <p className="text-black text-xl font-bold mt-6">{title}</p>
//             <p className="text-gray font-semibold">{artist}</p>
//         </div>
//     );
// };

// export default SongTitle;
