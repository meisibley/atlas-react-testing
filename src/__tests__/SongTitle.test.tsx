import React from 'react';
import { render } from '@testing-library/react';
import { test, expect } from 'vitest';
import { Song } from '../components/types';
import SongTitle from '../components/SongTitle';

test('renders correctly and matches snapshot', ()=> {
    const mockSong: Song = {
        id: 3,
        title: 'Test Song',
        artist: 'Test Artist',
        genre: 'Test genre',
        duration: '2:18',
        cover: 'Test https',
    };
    const { container } = render (
        <SongTitle title={mockSong.title} artist={mockSong.artist} />
    );

    expect(container).toMatchSnapshot();
})
