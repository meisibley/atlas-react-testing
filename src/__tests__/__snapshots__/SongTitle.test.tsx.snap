// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly and matches snapshot 1`] = `
<div>
  <div>
    <p
      class="text-black text-xl font-bold mt-6"
    >
      Test Song
    </p>
    <p
      class="text-gray font-semibold"
    >
      Test Artist
    </p>
  </div>
</div>
`;
